### PR TITLE
DTSPO-75 migrate to shared module

### PR DESCRIPTION
**Pull request by msl8r**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPO-429

### Change description ###
- Migrated to shared frontdoor module.
- Using log analytics workspace id module
- upgraded tf to 0.13 (will be upgrading to 14 once terraform has been applied to each environment)
- upgraded azurerm provider to 2.41.0
- ran terraform fmt to fix pester tests for 0.13 upgrade

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/global/provider.tf [hunk 1]
```diff
@@ -1,5 +1,17 @@
+terraform {
+  required_version = ">= 0.14.4"
+
+  backend "azurerm" {}
+  required_providers {
+    azurerm = {
+      source = "hashicorp/azurerm"
+      # https://github.com/terraform-providers/terraform-provider-azurerm/issues/8208
```
Comment (timja) [inline components/global/provider.tf:8]:
```suggestion
```

File: components/global/rg.tf [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
   name = "lz-${var.env}-rg"
+}
+
+data "azurerm_resource_group" "key_vault" {
+  name = var.env == "perftest" || var.env == "aat" || var.env == "preview" ? "core-infra-${var.subscription}-rg" : "core-infra-${var.env}-rg"
+}
+
+data "azurerm_log_analytics_workspace" "log" {
+  provider            = azurerm.loganalytics
+  name                = local.log_analytics_workspace[[for x in keys(local.log_analytics_env_mapping) : x if contains(local.log_analytics_env_mapping[x], var.env)][0]].name
```
Comment (timja) [inline components/global/rg.tf:11]: you can use the log analytics module for this?

https://github.com/hmcts/terraform-module-log-analytics-workspace-id/

would be nicer

File: components/global/local.tf [hunk 3]
```diff
@@ -31,4 +31,25 @@ locals {
...
+    }
+    nonprod = {
+      subscription_id = "1c4f0704-a29e-403d-b719-b90c34ef14c9"
+      name            = "hmcts-nonprod"
+    }
+    prod = {
+      subscription_id = "8999dec3-0104-4a27-94ee-6588559729d1"
+      name            = "hmcts-prod"
+    }
+  }
```
Comment (timja) [inline components/global/local.tf:54]:
```suggestion
```

File: components/global/main.tf [hunk 4]
```diff
@@ -19,4 +19,7 @@ module "landing_zone" {
   certificate_key_vault_name = var.certificate_key_vault_name
   oms_env                    = var.oms_env
   certificate_name_check     = var.certificate_name_check
+  key_vault_resource_group   = data.azurerm_resource_group.key_vault.name
+  log_analytics_workspace_id = data.azurerm_log_analytics_workspace.log.id
```
Comment (timja) [inline components/global/main.tf:23]: use the module instead https://github.com/hmcts/terraform-module-log-analytics-workspace-id/

File: components/global/provider.tf [hunk 5]
```diff
@@ -10,6 +22,8 @@ provider "azurerm" {
   subscription_id = var.data_subscription
 }
 
-terraform {
-  backend "azurerm" {}
+provider "azurerm" {
+  features {}
+  alias           = "loganalytics"
+  subscription_id = local.log_analytics_workspace[[for x in keys(local.log_analytics_env_mapping) : x if contains(local.log_analytics_env_mapping[x], var.env)][0]].subscription_id
 }
```
Comment (timja) [inline components/global/provider.tf:29]: shouldn't be needed
```suggestion
```

File: components/global/local.tf [hunk 6]
```diff
@@ -31,4 +31,26 @@ locals {
     "environment"        = local.env_display_names[var.env]
     "criticality"        = local.criticality[var.env]
   }
-}
+
```
Comment (msl8r) [inline components/global/local.tf:34]: only commenting out as I may need to revert for other envs

**Review by timja — APPROVED** [review]

**Pull request merged**